frontend/public/components/cluster-settings/openid-idp-form.tsx

```diff
@@ -0,0 +1,320 @@
+        });
+    });
+  };
+
+  oidTextValueChanged: React.ReactEventHandler<HTMLInputElement> = (event) => {
+    this.setState({[event.currentTarget.id]: event.currentTarget.value});
+  };
+
+  oidCBValueChanged: React.ReactEventHandler<HTMLInputElement> = event => {
+    this.setState({[event.currentTarget.id]: event.currentTarget.checked});
+  };
+
+  mappingMethodChanged = (mappingMethod: string) => {
+    this.setState({mappingMethod});
+  };
+
+  clientSecretFileChanged = (clientSecretFileContent: string) => {
+    this.setState({clientSecretFileContent});
+  };
+
+  caFileChanged = (caFileContent: string) => {
+    this.setState({caFileContent});
+  };
+
+  render() {
+    const { name, login, challenge, mappingMethod, clientID, clientSecretFileContent, preferredUserName, claimName, claimEmail, authorize, token, userInfo, caFileContent, extraScopes, extraAuthorizeParameters } = this.state;
+    const title = 'Add Identity Provider: OpenID Connect';
+    const mappingMethods = {
+      'claim': 'Claim',
+      'lookup': 'Lookup',
+      'add': 'Add',
+    };
+    return <div className="co-m-pane__body">
+      <Helmet>
+        <title>{title}</title>
+      </Helmet>
+      <form onSubmit={this.submit} name="form" className="co-m-pane__body-group co-m-pane__form">
+        <h1 className="co-m-pane__heading">{title}</h1>
+        <p className="co-m-pane__explanation">
+          Integrate with an OpenID Connect identity provider using an Authorization Code Flow.
+        </p>
+        <div className="form-group">
+          <label className="control-label co-required" htmlFor="name">Name</label>
+          <input className="form-control"
+            type="text"
+            onChange={this.oidTextValueChanged}
+            value={name}
+            aria-describedby="oid-name-help"
+            id="name"
+            required />
+          <p className="help-block" id="oid-name-help">
+            Unique name of the new identity provider. This cannot be changed later.
+          </p>
+        </div>
+        <div className="checkbox">
+          <label className="control-label">
+            <input type="checkbox" onChange={this.oidCBValueChanged} checked={login} id="login" />
+            Use this identity provider for browser login
+          </label>
+        </div>
+        <div className="checkbox">
+          <label className="control-label">
+            <input type="checkbox" onChange={this.oidCBValueChanged} checked={challenge} id="challenge" />
+            Issue <code>WWW-Authenticate</code> challenges to non-browser clients
+          </label>
+        </div>
+        <div className="form-group">
+          <label className="control-label co-required" htmlFor="tag">Mapping Method</label>
+          <Dropdown dropDownClassName="dropdown--full-width" items={mappingMethods} selectedKey={mappingMethod} title={mappingMethods[mappingMethod]} onChange={this.mappingMethodChanged} />
```

Comment:
We should eventually create a common component for mapping method so we don't repeat it 9 times, but I'd leave it for now